common/utils.go

```diff
@@ -35,7 +36,31 @@ func FormatValue(val any) any {
 	}
 }
 
-func NewCanal(config *mysql.Config, tables []string) (*canal.Canal, error) {
+// Canal is a wrapper around *canal.Canal to enforce our custom CdcPosition.
+type Canal struct{ *canal.Canal }
```

Comment:
Nitpick: I mean, this works, but it is a bit complicated IMO. I proposed to use explicit fields in `CdcPosition` when marshaling the position to JSON, so that's fine, but creating a layer of abstraction around the canal because of that is a bit much.